relentless/core/variable.py

```diff
@@ -0,0 +1,395 @@
+__all__ = ['Variable','DesignVariable','DependentVariable',
+           'UnaryOperator','SameAs',
+           'BinaryOperator','ArithmeticMean','GeometricMean'
+          ]
+
+import abc
+from enum import Enum
+
+import networkx as nx
+import numpy as np
+
+class Variable(abc.ABC):
+    """Abstract base class for any variable.
+
+    At minimum, a variable must provide the interface to get its value as a property.
+
+    """
+    @property
+    @abc.abstractmethod
+    def value(self):
+        pass
+
```

Comment:
Anticipating potential future needs to add other types of "independent" variables, or maybe so that we can try incorporate scalar constants into a dependent variable, can we add a light `IndependentVariable` class? It should derive from `Variable` and its constructor should take only one parameter `value`, which is stored in `self._value`. Its `value` property should just return `self._value`, and you can add a `value` setter as well that sets `self._value`.

Then, `DesignVariable` can derive from `IndependentVariable` and override these methods as necessary.